app/controllers/application_controller.rb

```diff
@@ -5,6 +5,151 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    set :sessions, true
+    set :method_override, true
+    set :session_secret, "fwitter_efrain"
   end
 
-end
\ No newline at end of file
+  get "/" do
+    erb :index
+  end
+
+  get "/signup" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :"users/create"
+    end
+  end
+
+  post "/signup" do
+
+    unless params[:username] == "" || params[:password] == "" || params[:email] == ""
+      User.create(username: params[:username], password: params[:password], email: params[:email])
+      session[:id] = User.last.id
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = all_tweets
+      erb :"tweets/tweets"
+    else
+      redirect '/login'
+    end
+  end
+
+  get "/tweets/new" do
+    if logged_in?
+      erb :"tweets/create"
+    else
+      redirect "/login"
+    end
+  end
+
+  post "/tweets" do
+    unless params[:content] == ""
+      current_user.tweets << Tweet.create(content: params[:content])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.